cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/RouteConfigHttpHandler.java

```diff
@@ -77,9 +82,22 @@ public void storeRouteConfig(HttpRequest request, HttpResponder responder,
                                @PathParam("namespace-id") String namespaceId,
                                @PathParam("app-id") String appId,
                                @PathParam("service-id") String serviceId) throws Exception {
-    ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
+    NamespaceId namespace = new NamespaceId(namespaceId);
+    ProgramId programId = namespace.app(appId).service(serviceId);
     Map<String, Integer> routes = parseBody(request, ROUTE_CONFIG_TYPE);
     RouteConfig routeConfig;
+    List<ProgramId> nonExistingServices = new ArrayList<>();
+    for (String version : routes.keySet()) {
+      ProgramId routeProgram = namespace.app(appId, version).service(serviceId);
+      if (lifecycleService.getProgramSpecification(routeProgram) == null) {
+        nonExistingServices.add(routeProgram);
```

Comment:
Should we just break the first time we see an non-existing service and send a response back?